### ISSUE TITLE
crash_generation_app deadlocks

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Pluu/google-breakpad#292]
```
What steps will reproduce the problem?
1. compile crash_generation_app in MSVC 2005 in debug mode
2. run crash_generation_app.exe on WinXP running on E6400 intel cpu
3. choose "Server->start"
4. choose "Client->Request Dump"

What is the expected output? What do you see instead?

I expect dump to be created and application not to freeze.
instead application freezes.

What version of the product are you using? On what operating system?

latest and greatest svn, which is r310 10 Feb 2009
Windows XP

Please provide any additional information below.

The dump created by the application is attached.
It shows that in the moment of dump creation the process was waiting 
for a semaphore on exception_handler.cc:593 

[WaitForSingleObject(handler_finish_semaphore_, INFINITE);]

after the application froze, I attached a debugger to it and saw that it is
still waiting on the same semaphore

```

Original issue reported on code.google.com by `avia...@gmail.com` on 17 Feb 2009 at 5:26
- Merged into: #273
